File: csharp/Chapter08/RunListClient.cs
using System;
using System.Collections.Generic;
using System.Text;
using System.Threading;
using com.db4o;

namespace com.db4o.dg2db4o.chapter8
{
    class RunListClient
    {
        ObjectContainer lClient = null;
        private bool stop = false;

        public void Run()
        {
            lock (this)
            {
                Console.WriteLine("Starting list client...");
                try
                {
                    lClient = Db4o.OpenClient("localhost", 8732, "user2", "password");
                    lClient.Set(new Person("Alice", 22));
                    lClient.Set(new Person("Bob", 26));
                    lClient.Commit();
                    while (!stop)
                    { // Your stop condition here...
                        ListResult(lClient.Get(new Person()));
                        Monitor.Wait(this, 10000);
                    }
                }
                catch (ThreadInterruptedException tie)
                {
                    Console.WriteLine("Thread Error!" + tie);
                }
                finally
                {
                    lClient.Close();
                }
            }
        }

        private static void ListResult(ObjectSet result)
        {
            Console.WriteLine("\nLISTCLIENT: Listing...");
            while (result.HasNext())
            {
                Console.WriteLine("LISTCLIENT:" + result.Next());
            }
        }

    }
}

File: java/src/com/db4o/dg2db4o/chapter8/RunListClient.java


package com.db4o.dg2db4o.chapter8;

import com.db4o.*;
import com.db4o.messaging.*;

public class RunListClient implements  Runnable{
    private boolean stop = false; // Use this flag to quit the
    ObjectContainer lClient = null;
    
    // Can run as standalone process, or use Main.java to run as thread
    public static void main(String[] args) {
        RunListClient runner = new RunListClient();
        runner.run();
    }
    
    public void run() {
        synchronized(this){
            try {
                lClient = Db4o.openClient("127.0.0.1", 8732, "user2", "password");
                while (!stop) {
                    listResult(lClient.get(new Person()));
                    Thread.sleep(15000);
                    //this.wait(15000);
                }
            } catch (Exception e) {
                e.printStackTrace();
            } finally {
                lClient.close();
            }
        }
    }
    
    public static void listResult(ObjectSet result) {
        System.out.println("LISTCLIENT: Listing...");
        while (result.hasNext()) {
            System.out.println("LISTCLIENT:" + result.next());
        }
    }
    
}
